Work with new Swift

tipabu opened the pull request:
Here we

* Add `X-Timestamp` headers before calling `constraints.check_object_creation` (like the object controller had to do in https://github.com/openstack/swift/commit/31c294d),
* Update the version of Swift we use when running unit tests to demonstrate why the above is necessary,
* Add more tox envs to test with both older and newer versions of Swift as well, which highlights why we can't use things like `Timestamp.now()`, and
* Add the pfs_middleware unit tests to the project-wide `make test` target.

tipabu committed:
Add something like _update_x_timestamp when creating objects

Otherwise, all PUTs to bimodal accounts will 400 following
https://github.com/openstack/swift/commit/31c294d

tipabu committed:
pfs_middleware: Update the version of swift we test against

tipabu committed:
pfs_middleware: Test against more versions of swift

The master job should be considered non-voting; sometimes Swift will break us.

tipabu committed:
Add pfs_middleware unit tests to makefile

smerritt reviewed (approved):
Looks good to me.

I sort of want CI to test against every Swift version we care about, but that can be a different set of changes.

Commit:
Add provisioning of python tox tool in build environments

Note new rule python-test added to ProxyFS/Makefile.
It is only run on Linux and, hence, you need tox there:

  sudo pip install tox

Re-provisioning should install tox for you...

edmc-ss commented:
I just pushed a small delta to Tim's PR that makes sure runway & SAIO can complete "make"...

Everything else looks great!
